src/main/java/org/elasticsearch/action/bulk/BulkRequest.java

```diff
@@ -285,41 +287,48 @@ public BulkRequest add(BytesReference data, @Nullable String defaultIndex, @Null
+                                routing = parser.text();
+                            } else if ("_parent".equals(currentFieldName) || "parent".equals(currentFieldName)) {
+                                parent = parser.text();
+                            } else if ("_timestamp".equals(currentFieldName) || "timestamp".equals(currentFieldName)) {
+                                timestamp = parser.text();
+                            } else if ("_ttl".equals(currentFieldName) || "ttl".equals(currentFieldName)) {
+                                if (parser.currentToken() == XContentParser.Token.VALUE_STRING) {
+                                    ttl = TimeValue.parseTimeValue(parser.text(), null).millis();
+                                } else {
+                                    ttl = parser.longValue();
+                                }
+                            } else if ("op_type".equals(currentFieldName) || "opType".equals(currentFieldName)) {
+                                opType = parser.text();
+                            } else if ("_version".equals(currentFieldName) || "version".equals(currentFieldName)) {
+                                version = parser.longValue();
+                            } else if ("_version_type".equals(currentFieldName) || "_versionType".equals(currentFieldName) || "version_type".equals(currentFieldName) || "versionType".equals(currentFieldName)) {
+                                versionType = VersionType.fromString(parser.text());
+                            } else if ("_retry_on_conflict".equals(currentFieldName) || "_retryOnConflict".equals(currentFieldName)) {
+                                retryOnConflict = parser.intValue();
                             } else {
-                                ttl = parser.longValue();
+                                throw new IllegalArgumentException("Action/metadata line [" + line + "] contains an unknown parameter [" + currentFieldName + "]");
                             }
-                        } else if ("op_type".equals(currentFieldName) || "opType".equals(currentFieldName)) {
-                            opType = parser.text();
-                        } else if ("_version".equals(currentFieldName) || "version".equals(currentFieldName)) {
-                            version = parser.longValue();
-                        } else if ("_version_type".equals(currentFieldName) || "_versionType".equals(currentFieldName) || "version_type".equals(currentFieldName) || "versionType".equals(currentFieldName)) {
-                            versionType = VersionType.fromString(parser.text());
-                        } else if ("_retry_on_conflict".equals(currentFieldName) || "_retryOnConflict".equals(currentFieldName)) {
-                            retryOnConflict = parser.intValue();
+                        } else {
+                            throw new IllegalArgumentException("Action/metadata line [" + line + "] has a wrong format");
```

Comment:
can we add some more info about what expected found and the field name? maybe "expected a simple value for field '..' but found a [TYPE]"..